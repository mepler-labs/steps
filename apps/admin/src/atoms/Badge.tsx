import React from 'react';
import styled from 'styled-components';

import {
  blue,
  brown,
  green,
  grey,
  pink,
  yellow,
  white,
  colorFromString,
} from 'styles/colors';

const badgeColors = [blue, brown, green, pink, yellow];

interface Props {
  className?: string;
  color?: string;
  text: string;
  rounded?: boolean;
}

const Badge: React.SFC<Props> = props => (
  <StyledBadge {...props}>{props.text}</StyledBadge>
);

const StyledBadge = styled<Props, 'div'>('div')`
  background-color: ${({ color, text }) => {
    if (color) return color;
    return colorFromString(text.toUpperCase(), badgeColors);
  }};
  border-radius: ${({ rounded }) => (rounded ? 1000 : 4)}px;
  color: ${white};
  display: inline-block;
  font-size: 0.825rem;
  padding-bottom: 0.4em;
  padding-left: 0.75em;
  padding-right: 0.75em;
  padding-top: 0.425em;
  text-transform: uppercase;
`;

export default Badge;
