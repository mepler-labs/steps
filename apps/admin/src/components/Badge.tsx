import React from 'react';
import styled from 'styled-components';

import { white, badgeColors } from '../styles/colors';

interface Props {
  className?: string;
  text: string;
}

class Badge extends React.Component<Props, {}> {
  render () {
    const { className, text } = this.props;

    return (
      <div className={className}>
        {text}
      </div>
    );
  }
}

const getColor = (text: string): string => {
  const colors = Object.values(badgeColors);

  let n = 0;
  for (var i = 0; i < text.length; ++i) {
    n = (n + text.charCodeAt(i)) % colors.length;
  }

  return colors[n];
};

const StyledLabel = styled(Badge)`
  background-color: ${props => getColor(props.text.toUpperCase())};
  border-radius: 3px;
  color: ${white};
  display: inline-block;
  padding-bottom: 0.325em;
  padding-left: 0.75em;
  padding-right: 0.75em;
  padding-top: 0.325em;
  text-transform: uppercase;
`;

export default StyledLabel;
