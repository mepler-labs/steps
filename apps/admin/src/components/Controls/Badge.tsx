import React from 'react';
import styled from 'styled-components';

import {
  blue,
  brown,
  green,
  grey,
  pink,
  yellow,
  white,
  colorFromString
} from 'styles/colors';

const badgeColors = [blue, brown, green, pink, yellow];

interface Props {
  className?: string;
  color?: string;
  text: string;
}

class Badge extends React.Component<Props, {}> {
  render() {
    const { className, text } = this.props;

    return <div className={className}>{text}</div>;
  }
}

const StyledBadge = styled(Badge)`
  background-color: ${({ color, text }) => {
    if (color) return color;
    return colorFromString(text.toUpperCase(), badgeColors)
  }};
  border-radius: 4px;
  color: ${white};
  display: inline-block;
  font-size: 0.825rem;
  padding-bottom: 0.4em;
  padding-left: 0.75em;
  padding-right: 0.75em;
  padding-top: 0.425em;
  text-transform: uppercase;
`;

export default StyledBadge;
