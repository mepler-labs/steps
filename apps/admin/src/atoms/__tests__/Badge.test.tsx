import Badge from '../Badge';
import { shallow } from 'enzyme';
import * as React from 'react';
import 'jest';

describe('Badge.tsx', () => {
  it('renders correctly', () => {
    const wrapper = shallow(
      <Badge text="badge text" className="my-class-name" />
    );

    expect(wrapper).toMatchSnapshot();
  });
});
