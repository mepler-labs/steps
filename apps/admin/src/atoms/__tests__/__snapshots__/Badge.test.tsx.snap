// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge.tsx renders correctly 1`] = `
<Badge
  className="my-class-name sc-bdVaJa iCCtii"
  text="badge text"
/>
`;
